dashboard/app/models/unit.rb

```diff
@@ -765,6 +757,25 @@ def self.latest_assigned_version(family_name, user)
       first
   end
 
+  # @param family_name [String] The family name for a unit family.
+  # @param user [User]
+  # @return [Unit|nil] Returns the latest unit version in a family that the user has progress in.
+  def self.latest_version_with_progress(family_name, user)
+    return nil unless family_name && user
+
+    family_unit_versions = Unit.where(family_name: family_name).
```

Comment:
this is not ideal because it hits the database. please copy the alternate approach from a few lines earlier, which hits the cache: https://github.com/code-dot-org/code-dot-org/blob/576bb6f9fdb066286e34bcd50f3d4e4ec97fa50d/dashboard/app/models/unit.rb#L723-L724

ideally `latest_assigned_version` would follow this approach too, but at the least we should follow the "don't make things worse" rule here.

Reply:
Good point! Will do!